ldai/client.go

```diff
@@ -1,12 +1,139 @@
 package ldai
 
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/alexkappa/mustache"
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/launchdarkly/go-sdk-common/v3/ldvalue"
+	"github.com/launchdarkly/go-server-sdk/ldai/internal/datamodel"
+	"github.com/launchdarkly/go-server-sdk/v7/interfaces"
+)
+
 type ServerSDK interface {
+	JSONVariation(
+		key string,
+		context ldcontext.Context,
+		defaultVal ldvalue.Value,
+	) (ldvalue.Value, error)
+	Loggers() interfaces.LDLoggers
+	TrackMetric(
+		eventName string,
+		context ldcontext.Context,
+		metricValue float64,
+		data ldvalue.Value,
+	) error
 }
 
 type Client struct {
-	sdk ServerSDK
+	sdk    ServerSDK
+	logger interfaces.LDLoggers
+}
+
+const ldContextVariable = "ldctx"
+
+func New(sdk ServerSDK) (*Client, error) {
+	if sdk == nil {
+		return nil, fmt.Errorf("sdk must not be nil")
+	}
+	return &Client{
+		sdk:    sdk,
+		logger: sdk.Loggers(),
+	}, nil
+}
+
+func (c *Client) Config(
+	key string,
+	context ldcontext.Context,
+	defaultValue Config,
+	variables map[string]interface{},
+) (Config, *Tracker) {
+	variation, err := c.sdk.JSONVariation(key, context, ldvalue.Null())
```

Comment:
Okay, so this is something we should talk about because I thought about doing this same thing.

You are passing in null here. In the python implementation, we are passing the default value through. In certain paths, you and I are emitting different shaped events since the default values are included in there.

We should probably figure out which one it should be.

Reply:
Ah, yea, I had left for the day before pushing up my latest change which passes in the marshalled default value. I think that is the most correct way to do this with the least amount of cleverness.